GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTCCTUploader.m

```diff
@@ -114,7 +114,9 @@ - (void)uploadPackage:(GDTUploadPackage *)package {
           [package completeDelivery];
           if (self->_backgroundID != GDTBackgroundIdentifierInvalid) {
             [[GDTApplication sharedApplication] endBackgroundTask:self->_backgroundID];
-            self->_backgroundID = GDTBackgroundIdentifierInvalid;
+            @synchronized(self) {
```

Comment:
We synchronize setting of `self->_backgroundID` but we don't synchronize its reading, so as far as I understand, the race condition is still possible. Will it be cleaner and safer if we use an `atomic` property to access `backgroundID`?

Reply:
This problem is perhaps better solved at the type declaration level, so I've removed the @synchs and made the type volatile.